cluster/docker-cluster/kube-deploy/destroy.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script contains useful functions which will be called during deployment
+# to clear bootstrap daemon and existing hyperkube containers.
+# And it is also used to tear down the whole deployment when calling kube-down. 
+
+# Clean bootstrap containers, and then destory the bootstrap daemon
+function clear_old_bootstrap {
+    echo "... Bootstrap daemon already started, destroying"
+    PID=`ps -eaf | grep 'unix:///var/run/docker-bootstrap.sock' | grep -v grep | awk '{print $2}'`
+
+    if [[ "" !=  "$PID" ]]; then
+        clear_bootstrap_containers
+        kill -9 $PID
+        echo "... Clearing bootstrap dir"
+        rm -rf /var/lib/docker-bootstrap || true
+        # Have to warn user some dirs left
+        echo "Warning: Some directories can not be deleted, you need to clear them mannually"
+    fi
+}
+
+function clear_bootstrap_containers {
+  # Clean the bootstrap containers
+  containers=`docker -H unix:///var/run/docker-bootstrap.sock ps -aq`
+  if [[ "" !=  "$containers" ]]; then
```

Comment:
This actually applies to many places. Either use ```-z``` or ```-n``` or just compare explicitly like here. But please be consistent.